sda/router.go

```diff
@@ -0,0 +1,550 @@
+package sda
+
+import (
+	"errors"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/network"
+	"github.com/dedis/crypto/abstract"
+	"golang.org/x/net/context"
+)
+
+// Router is an abstraction to represent the bridge between the communication
+// layer (network, channels etc) and the logical/processing layer (overlay &
+// protocols, services etc). It is a duplex communication link (send/receive)
+// from/to other router of the same type.
```

Comment:
"from/to other router of the same type." other router*s* or from/to another router (if it is just one)